Add toml configuration, rename startup commands and remove rpc node list

ferric-sol opened the pull request:
This PR adds a Firedancer-style hierarchical TOML config

    - Restructure config into sections: [ledger], [rpc], [replay], [development], [reporting], [overcast]
    - Use snake_case for all config keys matching Firedancer conventions
    - Keep 'rpc' and 'txpar' flag names unchanged
    - Fix CLI flag precedence: only override config when flag explicitly set

ferric-sol committed:
WIP: TOML config

ferric-sol committed:
match toml to frankendancer as best as we can

ferric-sol committed:
correctly map toml options to flags

ferric-sol committed:
refactor: Firedancer-style hierarchical TOML config

- Restructure config into sections: [ledger], [rpc], [replay], [development], [reporting], [overcast]
- Use snake_case for all config keys matching Firedancer conventions
- Keep 'rpc' and 'txpar' flag names unchanged as requested
- Fix CLI flag precedence: only override config when flag explicitly set

rupansh commented on cmd/mithril/main.go:
```diff
@@ -26,6 +27,9 @@ func init() {
 	cmd.PersistentFlags().AddGoFlagSet(klogFlags)
 
+	// Add config file flag
+	cmd.PersistentFlags().StringVar(&config.ConfigFile, "config", "", "Path to TOML config file")
```
Shall we set the default path to config.toml or smth?

ferric-sol replied:
added in https://github.com/ferric-sol/mithril/commit/2c33da0dddd56b2c1a74de8ffc7d5992fd27c913

rupansh commented on mithril.example.toml:
```diff
@@ -0,0 +1,124 @@
+    # rpc = "https://api.mainnet-beta.solana.com"
+
+    # Path to file containing list of RPC nodes
+    # rpc_node_list = "/path/to/rpcs.txt"
```
I think this should just be a list instead of reading it through a text file, and then we can remove the rpc flag as well

ferric-sol replied:
added in https://github.com/ferric-sol/mithril/commit/86469206bf6396accaacf790644171a88a7474b2

rupansh reviewed (changes requested):
This is mostly good! Just some minor tweaks

ferric-sol committed:
chore: remove mithril.toml from git, add to .gitignore

ferric-sol committed:
refactor: remove rpc_node_list, use single -r endpoint; default to config.toml

- Remove rpc_node_list/RpcPoolFile - single RPC endpoint now used for block downloads
- Default to config.toml if no --config flag provided
- Update instructions.md with new CLI flag names

ferric-sol committed:
feat: support multiple RPC endpoints via -r flag and config

- Change -r/rpc to accept comma-separated list of endpoints
- Block downloader uses all endpoints for load balancing
- Config rpc.rpc is now an array type

ferric-sol committed:
refactor: rename commands to verify-range/verify-live, add [block] config

- Rename verifier -> verify-range, catchup-rpc/catchup-overcast -> verify-live
- Add [block] section with source (rpc/overcast) and overcast_endpoint
- Merge RPC and Overcast catchup logic into single runVerifyLive()
- Falls back to RPC if source=overcast but no endpoint provided

ferric-sol committed:
fix: use configured RPC endpoints instead of hardcoded mainnet-beta

- Snapshot downloads now use rpcEndpoints[0] instead of hardcoded URL
- Block downloader uses pool client for GetSlot() calls
- Improve rate limit detection to catch '429' and 'Too many requests'
- Add log message showing which RPC endpoint is being used

ferric-sol committed:
fix: get CLI flag values directly instead of from viper

CLI flags weren't bound to viper, so getX() helpers were returning
empty values when CLI flags were set. Now reads flag values directly.

ferric-sol committed:
fix: respect flag default values when config key not set

getUint64, getInt64, getInt now fall back to flag's DefValue
when neither CLI flag nor config file sets the value. This fixes
BorrowedAccountArenas not being initialized (default 1024).

ferric-sol committed:
fix: add retry with exponential backoff for GetLeaderSchedule

Helius and other RPC providers rate-limit GetLeaderSchedule calls.
Now retries up to 10 times with exponential backoff (1s, 2s, 4s... max 30s).

ferric-sol committed:
fix: handle empty snapshot path from download library

Add check for empty finalPath after snapshot download attempt.
Return error instead of panicking with invalid filename format.

rupansh commented:
Looks good! Final signoff from @smcio
